query-engine/driver-adapters/js/adapter-neon/src/conversion.ts

```diff
@@ -95,3 +160,20 @@ types.setTypeParser(NeonColumnType.JSONB, convertJson)
 types.setTypeParser(NeonColumnType.JSON, convertJson)
 types.setTypeParser(NeonColumnType.MONEY, money => money.slice(1))
 types.setTypeParser(NeonColumnType.BYTEA, convertBytes)
+
+const parseBytesArray = types.getTypeParser(ArrayColumnType.BYTEA_ARRAY) as (_: string) => Buffer[]
+
+types.setTypeParser(ArrayColumnType.BYTEA_ARRAY, (serializedBytesArray) => {
+  const buffers = parseBytesArray(serializedBytesArray)
+  return buffers.map(encodeBuffer)
+})
+
+const parseTextArray = types.getTypeParser(ArrayColumnType.TEXT_ARRAY) as (_: string) => string[]
+
+types.setTypeParser(ArrayColumnType.TIME_ARRAY, parseTextArray)
+types.setTypeParser(ArrayColumnType.DATE_ARRAY, parseTextArray)
+types.setTypeParser(ArrayColumnType.TIMESTAMP_ARRAY, parseTextArray)
+
+types.setTypeParser(ArrayColumnType.MONEY_ARRAY, (moneyArray) =>
+  parseTextArray(moneyArray).map((money) => money.slice(1)),
+)
```

Comment:
Note: there is supposed to be `types.arrayParser` function that would've been super helpful for constructing these things but it only seems to exist in types but not at runtime in neon driver, so we have to make do with existing parsers we can retrieve with `type.getTypeParser`.